src/main/java/org/spongepowered/api/service/command/CommandService.java

```diff
@@ -44,72 +45,90 @@
      */
-    Optional<CommandMapping> register(Object plugin, CommandCallable callable, List<String> aliases);
+    Optional<CommandMapping> register(Object registrar, CommandCallable callable, List<String> aliases);
 
     /**
      * Register a given command using a given list of aliases.
      *
-     * <p>The provided callback function will be called with a list of aliases
-     * that are not taken (from the list of aliases that were requested) and
-     * it should return a list of aliases to actually register. Aliases may be
-     * removed, and if no aliases remain, then the command will not be
+     * <p>If the CommandService requires a single alias to only be registered
+     * once, the provided callback function will be called with a list of
+     * aliases that are not taken (from the list of aliases that were requested)
+     * and it should return a list of aliases to actually register. Aliases may
+     * be removed, and if no aliases remain, then the command will not be
      * registered. It may be possible that no aliases are available, and thus
      * the callback would receive an empty list. New aliases should not be added
      * to the list in the callback as this may cause
-     * {@link IllegalArgumentException} to be thrown.</p>
+     * {@link IllegalArgumentException} to be thrown. If the same alias can be
+     * registered multiple times in this CommandService, the callback will be
+     * ignored.</p>
      *
      * <p>The first non-conflicted alias becomes the "primary alias."</p>
      *
-     * @param plugin A plugin instance
+     * @param registrar The object registering the command. Can be a class
+     *        annotated by {@link Plugin}, or anything implementing
+     *        {@link CommandRegistrar}.
      * @param callable The command
      * @param aliases A list of aliases
      * @param callback The callback
-     * @return The registered command mapping, unless no aliases could be registered
-     * @throws IllegalArgumentException Thrown if new conflicting aliases are added in the callback
-     * @throws IllegalArgumentException Thrown if {@code plugin} is not a plugin instance
-     */
-    Optional<CommandMapping> register(Object plugin, CommandCallable callable, List<String> aliases, Function<List<String>, List<String>> callback);
-
-    /**
-     * Remove a mapping identified by the given alias.
-     *
-     * @param alias The alias
-     * @return The previous mapping associated with the alias, if one was found
+     * @return The registered command mapping, unless no aliases could be
+     *         registered
+     * @throws IllegalArgumentException Thrown if new conflicting aliases are
+     *         added in the callback
+     * @throws IllegalArgumentException Thrown if {@code plugin} is not a plugin
+     *         instance
+     * @throws IllegalArgumentException Thrown if a command name is invalid.
      */
-    Optional<CommandMapping> remove(String alias);
```

Comment:
@simon816 Because that would not be at all useful in some systems. SimpleCommandService supports it, but it would not make sense in SpongeCommandService, as you might remove multiple commands from multiple plugins at the same time.